examples/demo/Standard_Editors/CompoundEditor_demo.py

```diff
@@ -7,22 +7,21 @@
 This demo shows each of the four styles of the CompoundEditor
 """
 
-# Imports:
-from traits.api \
-    import HasTraits, Trait, Range
+from traits.api import Enum, HasTraits, Range, Union
 
-from traitsui.api \
-    import Item, Group, View
-
-# Define the demo class:
+from traitsui.api import Item, Group, View
 
 
+# Define the demo class:
 class CompoundEditorDemo(HasTraits):
     """ Defines the main CompoundEditor demo class.
     """
 
     # Define a compund trait to view:
-    compound_trait = Trait(1, Range(1, 6), 'a', 'b', 'c', 'd', 'e', 'f')
+    compound_trait = Union(
+        Range(1, 6), Enum('a', 'b', 'c', 'd', 'e', 'f'),
+        default_value=1
+    )
```

Comment:
It might be a bit harsh to have the script fails with a nasty exception when the user's traits version is less than 6.1, especially since TraitsUI hasn't changed its install requirement to Traits >= 6.1 yet (I think).

We could do something like this I guess:
```
def somename():
    try:
        from traits.api import Union
    except ImportError:
        # traits < 6.1
        from traits.api import Either
        return Either(Range(1, 6), Enum("a", "b", ...), default=1)
   else:
        return Union(Range(1, 6), Enum("a", "b", ...), default_value=1)
```
Would be a good example to highlight the difference between the two. The downside though is that one of the branches won't be tested with CI.
Alternatively, keep `Either` and add a comment about `Union` being better but require Traits > 6.1

Reply:
I think adding such a function might be too complex for simple demonstration of an editor, especially for people who are just starting out with Python and Traits. I changed it to `Either` and added a comment to inform about `Union`.